samples/amcache.py

```diff
@@ -213,6 +218,19 @@ def main(argv=None):
         w.writerow(["timestamp", "timestamp_type", "path", "sha1"])
         for e in sorted(entries, key=lambda e: e.timestamp):
             w.writerow([e.timestamp, e.type, e.entry.path, e.entry.sha1])
+
+    elif args.do_json:
+        for e in ee:
+            document = {}
+            for i in FIELDS:
+                document[i.name] = getattr(e, i.name)
+                if document[i.name] is None:
+                    document[i.name] = "-"
+                elif type(document[i.name]) == datetime.datetime:
+                    document[i.name] = str(document[i.name])
+
+            print(json.dumps(document, ensure_ascii=False).encode("utf-8"))
```

Comment:
the unicode handling here looks good to me. nice work! `json.dumps` returns either `str` or `unicode`, which you correctly encode into a specific representation. 

on windows, i've occasionally hit issues where stdout is open in text mode, so when writing binary data to stdout, the windows shell inserts some unexpected bytes. however, since here we're dealing with encoded text, i think we should be ok.